### PR TITLE
feat(savers): refetch opportunities on deposit/withdraw

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR refetches savers userData on deposit/withdraw, by waiting 15 seconds before both setting the Tx as completed and refetching. This way, we ensure that by the time the Tx is shown as completed, the position is updated.

## Notes

Refetching in the view-layer is only the temporary naive way to refetch until we build a THORChain savers parsers, which we will be able to now that we have memos. Fetching in `<Status />` effectively means that closing the status modal would not refetch, which could be the case for UTXOs with 5+ minutes to 1 confirmation, less likely for EVM chains / ATOM.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Testing

- Deposit/withdraw from savers
- Ensure the position is updated accordingly as soon as the Tx shows completed

## Screenshots (if applicable)

https://user-images.githubusercontent.com/17035424/215135254-19bea4b2-7371-4e63-a5d6-28bc157f6377.mov

https://user-images.githubusercontent.com/17035424/215135348-e3f17411-48ed-434e-950b-566f0b33c706.mov

https://user-images.githubusercontent.com/17035424/215135541-09f8359a-c39f-4125-9336-d6aa7b065eee.mov

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3701** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3701" target="_blank">Graphite</a> 
    * **PR #3702** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3702" target="_blank">Graphite</a> 
      * **PR #3703** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3703" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3703?utm_source=stack-comment).

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Status.tsx [hunk 1]
```diff
@@ -54,16 +59,38 @@ export const Status = () => {
...
-          usedGasFee: confirmedTransaction.fee?.value,
-        },
-      })
+    if (!accountId) return
+
+    if (confirmedTransaction && confirmedTransaction.status !== 'Pending' && contextDispatch) {
+      ;(async () => {
+        // Artificial longer completion time, since THORChain Txs take around 15s after confirmation to be picked in the API
+        // This way, we ensure "View Position" actually routes to the updated position
+        await new Promise(resolve => setTimeout(resolve, 17000))
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Status.tsx:68]: @reallybeard as polish we may want to add a bar/countdown or something, we can't avoid tx confirmation times, and 17s with no feedback is really long for users.

**Comment by 0xdef1cafe** [comment]
@gomesalexandre consider this stack up to 3707 approved, please feel free to merge on your own, i've reviewed them all.

this one obv has the most logic, and i agree a parser is the best, let's get this in and test functionally for now and we can refactor to the parser approach if this is an issue later. we're going to get the parser for "free" for displaying deposit/withdraw txs anyway.

**Comment by gomesalexandre** [comment]
Admin merging as effectively approved in https://github.com/shapeshift/web/pull/3703#issuecomment-1407214011

**Pull request merged**